### PR TITLE
wgsl: Add tests for operator precedence

### PR DESCRIPTION
**Pull request by jrprice**
Includes execution and validation tests.

<hr>

Not exhaustive, but should hit the key hazards.

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [X] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/execution/expression/precedence.spec.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import { GPUTest } from '../../../gpu_test.js';
+
+export const g = makeTestGroup(GPUTest);
+
+// The list of test cases and their expected results.
+interface Expression {
+  expr: string;
+  result: number;
+}
+const kExpressions: Record<string, Expression> = {
```
Comment (ben-clayton) [inline src/webgpu/shader/execution/expression/precedence.spec.ts:16]: Suggestions:
* Parameterize the test to no-op or strip out all whitespace.
* Extend the `decl` parameterization to replace the variable names with their literal value (`three` -> `3`).

**Review by ben-clayton — APPROVED** [review]
Approval with a couple of suggestions

File: src/webgpu/shader/execution/expression/precedence.spec.ts (hunk 1 above)
Comment (jrprice) [reply src/webgpu/shader/execution/expression/precedence.spec.ts:16]: Done.

**Review by ben-clayton — APPROVED** [review]
Nice. LGTM!

**Pull request merged**